python/ray/data/datasource/parquet_datasource.py

```diff
@@ -47,6 +47,15 @@
 FRAGMENTS_PER_META_FETCH = 6
 PARALLELIZE_META_FETCH_THRESHOLD = 24
 
+# The `num_cpus` for each metadata prefetching task.
+# Default to 0.5 instead of 1 because it is cheaper than normal read task.
+NUM_CPUS_FOR_META_FETCH_TASK = 0.5
+
+# The application-level exceptions to retry for metadata prefetching task.
+# Default to retry on `OSError` because AWS S3 would throw this transient
+# error when load is too high.
+RETRY_EXCEPTIONS_FOR_META_FETCH_TASK = [OSError]
```

Comment:
should we add this in general for all `FileBasedDatasource` reads? https://github.com/ray-project/ray/blob/97406423d8d6d093f52588006eb55624cbfc27cf/python/ray/data/datasource/file_based_datasource.py#L68

Reply:
Only Parquet has this metadata prefetching. So we can add it first here. For all file-based datasources, looks like the expoential backoff retry is working.